backend/pennyme/pennycollector.py

```diff
@@ -0,0 +1,326 @@
+"""Utils to parse pennycollector.com"""
+import logging
+from datetime import datetime
+from typing import Any, Dict, List, Optional, Tuple
+
+import bs4
+from googlemaps import Client as GoogleMaps
+
+from .locations import COUNTRY_TO_CODE, remove_html_and
+
+logger = logging.getLogger(__name__)
+
+WEBSITE_ROOT = "http://209.221.138.252/"
+AREA_PREFIX = WEBSITE_ROOT + "Locations.aspx?area="
+AREA_SITE = WEBSITE_ROOT + "AreaList.aspx"
+DATE = datetime.today()
+YEAR, MONTH, DAY = DATE.year, DATE.month, DATE.day
+UNAVAILABLE_MACHINE_STATES = ["Moved", "Gone", "Out of Order"]
+
+# StatesList
+def get_area_list_from_area_website(website) -> List[str]:
+    """
+    Get a list with areas from the overall area website.
+
+    Args:
+        website: AREA_SITE website.
+
+    Returns:
+        List[str]: A list of areas.
+    """
+
+    unparsed_locs = website.find("table", id="StatesList")
+    us_locs = [loc.split("\t")[-1] for loc in str(unparsed_locs).split("</a>")][
+        :-1
+    ]
+
+    non_us_unparsed = website.findAll("option", attrs={"selected": ""})
+    non_us_locations = []
+    for can in non_us_unparsed[1:]:
+        loc = str(can).split("</option>")[0].split(">")[-1]
+        if loc == "Select One":
+            break
+        non_us_locations.append(loc)
+    return us_locs + non_us_locations
+
+
+def validate_location_list(locations: List[str]) -> Tuple[bool, List[str]]:
+    """
+    Receive a list of locations (str) and validate that all of them are
+    known to the app.
+
+    Args:
+        List: List of locations
+
+    Returns:
+        Tuple of:
+            bool: Whether or not all locations are known.
+            diff: List of unknown locations (typically empty).
+    """
+    diff = [loc for loc in locations if loc not in COUNTRY_TO_CODE.keys()]
+    return diff == [], diff
+
+
+def get_raw_locations_from_location_website(website) -> List[bs4.element.Tag]:
+    """
+    Retrieves a list of locations from a website of any location, e.g.:
+    http://209.221.138.252/Locations.aspx?area=42
+
+    Args:
+        website: The bs4 website content.
+
+    Returns:
+         List[bs4.element.Tag]: List of bs4 tags
+    """
+    location_raw_table = website.find("table", attrs={"border": "1"})
+    location_raw_list = list(location_raw_table.find_all("td"))
+
+    # Skip the first 5 rows (they contain design issues)
+    return location_raw_list[5:]
+
+
+def get_location_list_from_location_website(website) -> List[List[str]]:
+    """
+    Retrieves a list of locations from a website of any location, e.g.:
+    http://209.221.138.252/Locations.aspx?area=42
+
+    Args:
+        website: The bs4 website content.
+
+    Returns:
+        List[List[str]]: List of locations, each represented as a list of strings,
+            one per column.
+    """
+    raw_locations = get_raw_locations_from_location_website(website)
+
+    # 5 Tags always make up one location
```

Comment:
I don't really get this part, what kind of "tags"

Reply:
The 5 columns here: http://209.221.138.252/Locations.aspx?area=42